src/js/_enqueues/wp/wp-tooltip.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Tooltip functionality for WordPress.
+ *
+ * @version 6.7.0
+ * @output wp-includes/js/wp-tooltip.js
+ */
+
+document.addEventListener( 'DOMContentLoaded', function () {
+	var tooltipContainers = document.querySelectorAll(
+		'.wp-tooltip-container'
+	);
+
+	tooltipContainers.forEach( function ( tooltipContainer ) {
+		var tooltipButton =
+			tooltipContainer.querySelector( '.wp-tooltip-button' );
+		var tooltipContent = tooltipContainer.querySelector(
+			'.wp-tooltip-content'
+		);
+
+		function showTooltip() {
+			tooltipContent.style.display = 'block';
+			adjustTooltipPosition( tooltipContainer, tooltipContent );
+		}
+
+		function hideTooltip() {
+			tooltipContent.style.display = 'none';
+		}
+
+		// Event listeners for mouse and touch events
+		tooltipContainer.addEventListener( 'mouseenter', showTooltip );
+		tooltipContainer.addEventListener( 'mouseleave', hideTooltip );
```

Comment:
For accessibility, the tooltip should not be hidden on `mouseleave`; only when either the trigger loses focus or when `esc` is hit.